### PR TITLE
feat: add validation for empty options while setting two factor auth

### PR DESCRIPTION
**Pull request by lucascnunes**
Adding validation to the form while setting up the Two Factor Authentication, if the user doesn't select any option, was returning an error.

Captura de Tela 2024-10-12 às 13 29 10

### CODE REVIEW

File: src/Pages/TwoFactor.php [hunk 1]
```diff
@@ -193,11 +193,21 @@ public function enableAction(): Action
...
             }
 
-            if ($this->twoFactorData['option']) {
-                $formData['two_factor_type'] = TwoFactorType::tryFrom($this->twoFactorData['option']);
+            try {
+                if ($this->twoFactorData['option']) {
+                    $formData['two_factor_type'] = TwoFactorType::tryFrom($this->twoFactorData['option']);
+                }
+            } catch (\Exception $e) {
+                Notification::make()
```
Comment (Baspa) [inline src/Pages/TwoFactor.php:201]: I think it would be better to show the message in the form itself like a validation error. Can you do that maybe?

**Comment by lucascnunes** [comment]
What do you think of this approach? Defining a default value from the beginning avoiding the null value error?

File: src/Pages/TwoFactor.php (hunk 1 above)
Comment (lucascnunes) [reply src/Pages/TwoFactor.php]: I've changed the code, check it out.

Comment (lucascnunes) [reply src/Pages/TwoFactor.php]: I'm totally new with filamentphp, haven't found a way of returning the error to the form yet, if this new approach isn't the best solution, I will continue to search for.

**Comment by Baspa** [comment]
> What do you think of this approach? Defining a default value from the beginning avoiding the null value error?

Excuse me for the late response, I was on a holiday. I will check it asap.

**Comment by Baspa** [comment]
Is it possible for you to fix the PHPStan errors? The solutions seems good to me @lucascnunes

**Comment by Baspa** [comment]
Seems like these errors are still happening @lucascnunes: 

> Error: Unable to resolve the template type TKey in call to function collect
Error: Unable to resolve the template type TValue in call to function collect
Error: Ternary operator condition is always true.
Error: Process completed with exit code 1.

**Comment by lucascnunes** [comment]
I'm having difficulties to test it locally,
When I run locally it gives to errors, what am I missing? 

Screenshot 2024-11-13 at 14 52 21

**Comment by Baspa** [comment]
Seems like you didn't commit changes, I don't see any changes in your last commit @lucascnunes

**Comment by lucascnunes** [comment]
Finally!

**Comment by Baspa** [comment]
> Finally! 😅
> 
> I've never used PHPStan, have no idea what type of error was that.
> 
> If it's all correct, I'm glad I could help a little bit, thanks for the opportunity.

Thanks! I will merge the PR

**Pull request merged**